lib/request-handler.js

```diff
@@ -16,20 +16,22 @@ var retryOnNextHostDecision = {
 
 /**
  * Handles a request to Cassandra, dealing with host fail-over and retries on error
- * @param {?Client} client Client instance used to retrieve and set the keyspace. It can be null.
+ * @param {Client} client Client instance used to retrieve and set the keyspace.
  * @param {Object} options
  * @constructor
  */
 function RequestHandler(client, options) {
   this.client = client;
-  this.loadBalancingPolicy = options.policies.loadBalancing;
-  this.retryPolicy = options.policies.retry;
+  this.profileManager = client.profileManager;
+  var profile = client.profileManager.getDefault();
+  this.loadBalancingPolicy = profile.loadBalancing;
+  this.retryPolicy = profile.retry;
```

Comment:
Another thing I was wondering is why the retry policy can be specified on the query options, but not the load balancing policy?   My guess:   The load balancing policies are used in control connection for determining host distance, so if the user were to provide an lbPolicy in the query options, it could violate the distance resolution there.